src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosResponseHandler.cs

```diff
@@ -60,6 +60,31 @@ public override async Task<ResponseMessage> SendAsync(RequestMessage request, Ca
 
         private void UpdateOptions(RequestMessage options)
         {
+            IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.RequestContext;
+            if (fhirRequestContext == null)
+            {
+                return;
+            }
+
+            _cosmosDataStoreConfiguration.ContinuationTokenSizeLimitInKb = Constants.ContinuationTokenDefaultLimit;
```

Comment:
I think `_cosmosDataStoreConfiguration` is a singleton, so this is likely going to affect all requests

Reply:
@brendankowitz  I fixed the code not to touch _cosmosDataStoreConfiguration.ContinuationTokenSizeLimitInKb.
Code now looks in the header and use the token size if present else tries to use the value from _cosmosDataStoreConfiguration.ContinuationTokenSizeLimitInKb.